card/mod/05_standard/set/all/rich_html/form.rb

```diff
@@ -25,11 +25,14 @@ def edit_slot args={}
 
 
   def form_for_multi
-    block = Proc.new {}
-    builder = ActionView::Base.default_form_builder
     card.name = card.name.gsub(/^#{Regexp.escape(root.card.name)}\+/, '+') if root.card.new_card?  ##FIXME -- need to match other relative inclusions.
 
-    builder.new("card[subcards][#{card.relative_name}]", card, template, {}, block)
+    # doesn't work anymore in Rails 4
+    # TODO -- check whether forms work with the new instantiate_builder call
+    # block = Proc.new {}
+    # builder = ActionView::Base.default_form_builder
+    # builder.new("card[subcards][#{card.relative_name}]", card, template, {}, block)
+    builder = instantiate_builder("card[subcards][#{card.relative_name}]", card, {})
```

Comment:
Well, template was/is a wart.  It just called this:

````
    def template
      @template ||= begin
        c = controller
        t = ActionView::Base.new c.class.view_paths, {:_routes=>c._routes}, c
        t.extend c.class._helpers
        t
      end
    end
````

I think we had to do this to get access to all the standard rails form API.

We should probably come up with a cleaner idea of how we want people to create views.  We've got a lot of divergent models now (rails helpers, haml, do-it-yourself...)